### PR TITLE
fix: DIRECTORY_SCOPE option.

### PR DESCRIPTION
**Pull request by cyfdecyf**
Setting variables in parent scope means the current scope is not affected. Thus the previous `DIRECTORY_SCOPE` has not effect.

This PR also makes `XMAKE_CMD` a user settable variable, along with updated README.

### CODE REVIEW

**Comment by waruqi** [comment]
xmake not found on ci

File: xrepo.cmake [hunk 1]
```diff
@@ -281,24 +291,29 @@ macro(_xrepo_fetch_json)
...
     endif()
 
     if(DEFINED linkdirs)
+        set(${package_name}_LINK_DIR "${linkdirs}")
         set(${package_name}_LINK_DIR "${linkdirs}" PARENT_SCOPE)
-        message(STATUS "xrepo: ${package_name}_LINK_DIR ${linkdirs}")
+        message(STATUS "xrepo: ${package_name}_LINK_DIR ${${package_name}_LINK_DIR}")
     else()
         message(STATUS "xrepo fetch --json: ${package_name} linkdirs not found")
     endif()
```
Comment (waruqi) [inline xrepo.cmake:319]: How can I add libraries? e.g. `-lfoo -lbar`

I only see the parsing of includedirs and linkdirs, but not the link name of the parsing library.

Comment (cyfdecyf) [reply xrepo.cmake:319]: There are wo cases:

- For libraries providing `find_package`, user should link to the target exposed in `find_package`
  - `xrepo_package` sets variable `<pkg>_DIR` thus the user just call `find_package` as usual
- For libraries without `find_package`, the user should figure this out by themselfs
  - This is how the user is using the library if they are not using `xrepo_package`
  - Personally, I've never called `pkg-config` to get link libraries in CMake before, so I guess this is what most users are doing

Comment (waruqi) [reply xrepo.cmake:319]: I have add `xrepo_target_packages` to solve this issue.

Comment (cyfdecyf) [reply xrepo.cmake:319]: That's really nice. I don't do cross platform dev a lot, didn't thought about this.

**Pull request merged**